test/src/gatherers/accessibility.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* eslint-env mocha */
+
+const AccessibilityGather = require('../../../src/gatherers/accessibility');
+const assert = require('assert');
+let accessibilityGather;
+
+const isExpectedOutput = artifact => {
+  return 'raw' in artifact && 'value' in artifact;
+};
+
+describe('Accessibility gatherer', () => {
+  // Reset the Gatherer before each test.
+  beforeEach(() => {
+    accessibilityGather = new AccessibilityGather();
+  });
+
+  it('fails gracefully', () => {
+    return accessibilityGather.postProfiling({
+      driver: {
+        evaluateAsync() {
+          return Promise.resolve();
+        }
+      }
+    }).then(_ => {
+      assert.ok(typeof accessibilityGather.artifact === 'object');
+    });
+  });
+
+  it('handles driver failure', () => {
+    return accessibilityGather.postProfiling({
+      driver: {
+        evaluateAsync() {
+          return Promise.reject('such a fail');
+        }
+      }
+    }).then(_ => {
+      assert(false);
```

Comment:
this assert it thrown away because of the catch below, so the real check is the `isExpectedOutput` below. You could just drop the `then` or you could do what's done in the module index.js and do the [`onFulfilled`/`onRejected` form to handle both branches](https://github.com/GoogleChrome/lighthouse/blob/d5ecf9933069d628c036bfa1f15a5f1807559289/test/src/index.js#L60)